Playmate/View Controllers/Home Tab/Sessions Related/SessionDetailsViewController.m

```diff
@@ -274,23 +276,23 @@ - (IBAction)addMyself:(id)sender {
         [self enableInviteButton];
         
         PFQuery *query = [PFQuery queryWithClassName:@"SportsSession"];
+        Session *session = [query getObjectWithId:self.sessionDetails.objectId];
 
-        // Retrieve the object by id
-        [query getObjectInBackgroundWithId:self.sessionDetails.objectId
-                                     block:^(PFObject *session, NSError *error) {
-            // Add myself to the session
-            NSMutableArray *oldPlayersList = (NSMutableArray *)session[@"playersList"];
-            [oldPlayersList addObject:me];
-            
-            session[@"playersList"] = (NSArray *)oldPlayersList;
-            
-            int newOccupied = (int)oldPlayersList.count;
-            session[@"occupied"] = [NSNumber numberWithInt:newOccupied];
-            
-            [session saveInBackground];
-        }];
+        NSMutableArray *oldPlayersList = (NSMutableArray *)session[@"playersList"];
+        [oldPlayersList addObject:me];
         
-        NSString *sessionObjectId = self.sessionDetails.objectId;
+        session[@"playersList"] = (NSArray *)oldPlayersList;
+        
+        int newOccupied = (int)oldPlayersList.count;
+        session[@"occupied"] = [NSNumber numberWithInt:newOccupied];
+        
+        [session saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+            if (succeeded) {
+                [self returnToHomeButStay:YES];
+            } else {
+                [Helpers handleAlert:error withTitle:[Strings errorString] withMessage:nil forViewController:self];
+            }
+        }];
```

Comment:
If these lines are exactly the same as 251-257 is it worth to put them on a separate method? something like `saveInBackgroundWithBlock` ?